src/databricks/labs/ucx/framework/parallel.py

```diff
@@ -27,8 +27,10 @@ def __init__(self, name, tasks: list[Callable[..., Result]], num_threads: int):
         self._default_log_every = 100
 
     @classmethod
-    def gather(cls, name: str, tasks: list[Callable[..., Result]]) -> (list[Result], list[Exception]):
-        num_threads = os.cpu_count() * 2
+    def gather(cls, name: str, tasks: list[Callable[..., Result]], num_threads: int = None) \
+            -> (list[Result], list[Exception]):
+        if num_threads is None:
+            num_threads = os.cpu_count() * 2
```

Comment:
next the following two lines into this if statement - otherwise the value would always be 8